tests/transformations/test_ChannelsLast.py

```diff
@@ -0,0 +1,214 @@
+import pytest
+
+import numpy as np
+import urllib.request
+
+import finn.core.onnx_exec as oxe
+from finn.core.modelwrapper import ModelWrapper
+from finn.custom_op.registry import getCustomOp
+from finn.transformation.general import GiveUniqueNodeNames
+from finn.transformation.infer_shapes import InferShapes
+from finn.util.basic import is_finn_op
+from qonnx.custom_op import ChannelsLast
+from qonnx.transformation.channelsLast import (
+    AbsorbChanFirstIntoMatMul,
+    ConvertToChannelsLastAndClean,
+    InsertChannelsLastDomainsAndTrafos,
+    MoveChanFirstDownstream,
+    MoveChanLastUpstream,
+    RemoveConsecutiveChanFirstAndChanLastTrafos,
+)
+from qonnx.util.cleanup import cleanup
+
+
+def download_model(test_model):
+    if test_model == "FINN-CNV_W2A2":
+        qonnx_url = "https://drive.google.com/uc?id=1BAyYM3N28fDHSWMgK8dZ3kVUsxGZLO3U&export=download"
+
+    elif "RadioML_VGG10":
+        qonnx_url = (
+            "https://github.com/Xilinx/brevitas-radioml-challenge-21/raw/"
+            "9eef6a2417d6a0c078bfcc3a4dc95033739c5550/sandbox/notebooks/models/pretrained_VGG10_w8a8_20_export.onnx"
+        )
+    else:
+        raise ValueError(f"Model called {test_model} is not supported")
+
+    # download test data
+    dl_dir = "/tmp"
+    dl_file = dl_dir + f"/{test_model}.onnx"
+    urllib.request.urlretrieve(qonnx_url, dl_file)
+    # run cleanup with default settings
+    out_file = dl_dir + f"/{test_model}_clean.onnx"
+    cleanup(dl_file, out_file=out_file)
+    return out_file
+
+
+def get_golden_in_and_output(onnx_file, test_model):
+    if test_model == "FINN-CNV_W2A2":
+        input_shape = (1, 3, 32, 32)
+    elif "RadioML_VGG10":
+        input_shape = (1, 2, 1024)
+    else:
+        raise ValueError(f"Model called {test_model} is not supported")
+    input_tensor = np.ones(np.prod(np.asarray(input_shape)), dtype=np.float32)
+    input_tensor = input_tensor.reshape(input_shape)
+
+    model = ModelWrapper(onnx_file)
+    input_dict = {model.graph.input[0].name: input_tensor}
+    golden_output_dict = oxe.execute_onnx(model, input_dict, True)
```

Comment:
do you need the `True` here? (returns full execution context)

Reply:
Yeah, it's not really needed, so I removed it.